src/values/list.rs

```diff
@@ -19,13 +19,15 @@ use values::*;
 
 pub struct List {
     frozen: bool,
```

Comment:
I'd be tempted to update this to be a tri-state enum:
```rust
enum Mutability {
  Mutable,
  Immutable,
  FrozenForIteration,
}
```

rather than having two inter-dependent bools - the fact that unfreezing for iteration noops if the thing has been frozen is a little more complex/confusing to me than just having a straight state transition, but fine as-is too.

Reply:
I actually was tempted to do so too but got lazy, I'll do it :)

Reply:
This is now done, can you take another look?